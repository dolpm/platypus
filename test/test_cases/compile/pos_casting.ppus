pipe main [] |> unit {
  int x = -5;
  float y = Float <| [-x-500];
  
  int z = Int <| [%y];

  Printnl <| [x];
  Printnl <| [y];
  Printnl <| [z];


  bool t = true;
  int p = Int <| [t];

  char c = 'C';

  float centipede = Float <| [Int <| [c]];

  Printnl <| [centipede];

  str yamother = Str <| [17.38];

  str i_thought_about_a_cloud = Str_new <| [" i'm like hey what's up hello!"];

  str astaghfirullah = yamother ^ i_thought_about_a_cloud;

  Printnl <| [&astaghfirullah];

  bool seeta_meeta_thrigma = false;
  str booled_up = Str <| [!seeta_meeta_thrigma];
  Printnl <| [&booled_up];

  |> ();
}
