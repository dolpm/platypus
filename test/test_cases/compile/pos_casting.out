-5
-495
-495
67
17.38 i'm like hey what's up hello!
